### PR TITLE
fix: trigger validation, name and version combined

### LINKED ISSUE #26: missing indication for config validation failures in monaco
when creating a new config, its not obvious whats wrong with the config at a glance.
we need to improve the indication and visibility of validation failures.

### LINKED ISSUE #42: Trigger validation Error on component mount.
On step 2 of creating new config, trigger the validation process on component mount.

### LINKED ISSUE #45: Configs Table: Combine Name and Version
- Updating the `Configs` table to combine the `display name` and `version` into a single column.

### PR DESCRIPTION
**Pull request by almog8k**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✔                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✖                                                                        |
| Chore            | ✖                                                                       |

Further  information:
- More details on create config step-2
- Combining name and version in configs table
-  Trigger validation on step-2 component mount

closes #45 
closes #42 
closes #26

### CODE REVIEW

**Comment by CptSchnitz** [comment]
ill review this after you merge the other PR. it has changes from both mixed inside

File: src/pages/configs/configTable/configTable.tsx [hunk 1]
```diff
@@ -17,22 +17,22 @@ type ConfigTableProps = {
...
-          <Box className={Styles.configNameColumn}>
+          <Box className={Styles.columnWithCopyButton}>
+            <ClipboardCopyButton text={row.configName} />
+
             <MuiLink underline="none" component={Link} to={`/config/${row.configName}/${row.version}`}>
-              <Tooltip title={row.configName} placement="top-start">
+              <Tooltip title={`${row.configName}:v${row.version}`} placement="top-start">
                 <Typography className={Styles.truncate} noWrap>
-                  {row.configName}
+                  {`${row.configName}:v${row.version}`}
```
[the remark this answers is not in the document]
Comment (almog8k) [reply src/pages/configs/configTable/configTable.tsx:32]: done

File: src/pages/createConfig/step2AddConfig/step2AddConfig.tsx [hunk 2]
```diff
@@ -1,24 +1,25 @@
...
+import { Link } from 'react-router-dom';
 
 type Step2AddConfigProps = {
   schemaId: string;
   onDataChange: (data: ConfigData | undefined, isValid: boolean) => void;
   onJsonStringChange: (json: string | undefined) => void;
   initialJsonStringData?: string | undefined;
 };
-export const Step2AddConfig: React.FC<Step2AddConfigProps> = ({ onDataChange, onJsonStringChange, schemaId, initialJsonStringData }) => {
+export const Step2AddConfig: React.FC<Step2AddConfigProps> = ({ onDataChange, onJsonStringChange, schemaId, initialJsonStringData = '{}' }) => {
```
[the remark this answers is not in the document]
Comment (almog8k) [reply src/pages/createConfig/step2AddConfig/step2AddConfig.tsx:20]: done

File: src/pages/createConfig/step2AddConfig/step2AddConfig.tsx [hunk 3]
```diff
@@ -58,35 +63,76 @@ export const Step2AddConfig: React.FC<Step2AddConfigProps> = ({ onDataChange, on
...
       }
     } catch (error) {
       let errMessage = 'Error parsing JSON';
-      if (error instanceof Error) {
+      if (error instanceof SyntaxError) {
+        console.log('error', error);
         errMessage = error.message;
       }
-      newErrors.push({ title: ErrorType.JSON_PARSE_ERROR, errorMessage: errMessage });
+      console.log('value', value);
```
[the remark this answers is not in the document]
Comment (almog8k) [reply src/pages/createConfig/step2AddConfig/step2AddConfig.tsx:79]: removed

File: src/pages/createConfig/step2AddConfig/step2AddConfig.tsx [hunk 4]
```diff
@@ -31,10 +32,14 @@ export const Step2AddConfig: React.FC<Step2AddConfigProps> = ({ onDataChange, on
   });
 
   const [isFetching, setIsFetching] = useState<boolean>(false);
-  const [errors, setErrors] = useState<ErrorCardProps[]>([]);
+  const [errors, setErrors] = useState<DetailCardProps[]>([]);
+
+  useEffect(() => {
+    handleEditorChange(initialJsonStringData);
+  }, [isSuccess]);
```
[the remark this answers is not in the document]
Comment (almog8k) [reply src/pages/createConfig/step2AddConfig/step2AddConfig.tsx:39]: tried it, it's crushing the component

**Review by CptSchnitz — APPROVED** [review]

**Pull request merged**